### ISSUE TITLE
Virginia data is messed up for 2020

### ISSUE THREAD

**Issue by hotdogPi** [khwilson/RedrawStates#11]
This issue only applies to 2020, not 2016.

There are three cities (that I know of) in Virginia that share the name of a county: Richmond, Roanoke, and Fairfax. In all three cases, the webpage uses the county numbers for both the county and the city. This turns Richmond from a big blue city to a tiny red one, giving Trump about 75,000 votes. It turns Roanoke from a blue city to a red city without changing its size much, giving Trump about 30,000 votes. The biggest issue is with Fairfax. While both are blue, Fairfax County is huge, but the City of Fairfax is pretty small. Using the Fairfax County data for Fairfax City gives Biden an extra 250,000 votes, which is huge. In addition, Fairfax City is unclickable, so you can't do anything with these almost 600,000 phantom votes.

Unrelated, I also want to note that your 2016 data is still preliminary despite being 4 years ago.

**Comment by khwilson** [comment]
Oh Virginia! Thank you for catching this. I had a different source for data last time and had to standardize them by hand this time. I was bad and didn't merge the code into one repo yet. The issue is here:

https://github.com/khwilson/RedrawStates2020/blob/main/redraw/cli.py#L126

As for whether or not Fairfax City is clickable, do you have any suggestions for that? In 2016, my solution was to tell people to zoom in, which is less than ideal. :-)

Finally, if you know of a good source for 2016 data, then I'm very happy to use it instead. The vote tabulations finished later than I made the tool, and at the time there were no easy APIs or github repos filled with county level data. What a difference four years makes!

**Comment by hotdogPi** [comment]
Some cities in Virginia are legitimately unselectable.

Perfectly normal: Alexandria, Bristol, Chesapeake, Danville, Hampton, Lynchburg, Newport News, Norfolk, Petersburg, Poquoson, Portsmouth, Richmond, Suffolk, Virginia Beach, Williamsburg

These cities show as a line instead of a polygon:

Galax: Galax and the two adjacent counties must all be in the same state for Galax to be selectable. The same is true for Fredericksburg and Falls Church.

Salem: Roanoke County and Salem must be in the same state for Salem to be selectable. Unlike the above three, it doesn't matter which state Roanoke City is in.

Manassas Park: Manassas Park and Prince William County must both be in Virginia (not any other state) for Manassas Park to be selectable. This is likely because of unselectable Manassas nearby.

Franklin: Franklin and Isle of Wight County must be in Virginia the same state for Franklin to be selectable. It doesn't matter which state Southampton County is in.

Fredericksburg: Galax and the two adjacent counties must all be in the same state for Galax to be selectable.

All independent cities completely surrounded by a county (Norton, Covington, Lexington, Staunton, Waynesboro, Harrisonburg, Charlottesville, Winchester, Emporia, Martinsville, Fairfax City) are completely unselectable, and they don't show on the map at all. Manassas and Colonial Heights are also completely unselectable, but moving all adjacent cities and counties to another state will reveal a thin black line that can't be interacted with. Hopewell is unselectable but does not show this thin black line despite being on the border of two counties.

**Comment by khwilson** [comment]
You're totally right! Over Thanksgiving, I reworked the 2016 and 2020 data pulls to all use the New York Times and to use a 10m-based topojson from the us-atlas repo. It seems that in the final topojson quantization, some of the super tiny cities get rounded down to "empty geographies" :-/

I think the solution to Virginia is to just legitimately replace the entirety of Virginia in the topojson with, e.g., the New York Times Virginia representation. https://static01.nyt.com/elections-assets/map-data/2020/v1/counties/VA_counties.json

However, given the way topojson works, there may be issues at boundaries. I can try to take a crack at it a little later today. But if you want to take a crack, I think the new version has a very simplified process for getting data. :-)

**Comment by hotdogPi** [comment]
https://kevinhayeswilson.com/redraw/ <https://kevinhayeswilson.com/redraw/> is still showing the old data for me, including the Richmond city/county error.

**Comment by khwilson** [comment]
Ah, I hadn't pushed to the public site yet. Need to copy / paste. The CI /
CD for this repo is not good. I'd been working on the dev branch by running
`cd public && python3 -m http.server`.

Anyway, should be up now!

**Comment by adamfilinovich** [comment]
Lexington VA isn't visible, much less selectable
